qiskit/transpiler/passes/optimization/consolidate_blocks.py

```diff
@@ -27,6 +29,7 @@
     "cz": CZGate(),
     "iswap": iSwapGate(),
     "ecr": ECRGate(),
+    "rxx": RXXGate(pi / 2),
```

Comment:
why is the angle here fixed to pi/2 ?

Reply:
It's because that's what `UnitarySynthesis` uses if it's a `basis_gates` only setup: https://github.com/Qiskit/qiskit/blob/main/qiskit/transpiler/passes/synthesis/unitary_synthesis.py#L98 the idea is that it's trying to match the decomposer used for estimating to what the synthesis pass will use later.

The whole logic between the passes is more than a bit hacky. `ConsolidateBlocks` tries to guess how many 2q gates will be used to decide whether to consolidate the block into a unitary or not. This is strongly coupled to the behavior of `UnitarySynthesis` because that's what actually does the synthesis later. But none of this actually takes into account multiple possible decomposers/synthesis methods being supported by a target, or the presence of a non-default plugin being used.

Reply:
FWIW, I did some testing locally with this and in some cases using `TwoQubitBasisDecomposer(RXXGate(pi/2))` under estimates what the `XXDecomposer` estimates, 3 vs 4 gates. This might result in some blocks not getting consolidated which were before, but in practice I find it probably pretty unlikely since it only counts rzz gates when evaluating if the block's gates are more than the expected synthesis output.